From 323c50b1477a306a648df2b592abb4e513568250 Mon Sep 17 00:00:00 2001
From: Roy Pledge <Roy.Pledge@freescale.com>
Date: Wed, 24 Apr 2013 13:59:32 -0400
Subject: [PATCH 279/547] Fix USDPAA memory mappings

Fix USDPAA memory mappings so they can no longer occur at
address 0 which confused any SW checking for NULL pointers.

Change-Id: Ic0ac5e61f1776ce8d4ddb622e401942ce2750705
Reviewed-on: http://git.am.freescale.net:8181/2062
Reviewed-by: Thorpe Geoff-R01361 <Geoff.Thorpe@freescale.com>
Reviewed-by: Ladouceur Jeffrey-R11498 <Jeffrey.Ladouceur@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
[Original patch taken from QorIQ-SDK-V1.4-SOURCE-20130625-yocto.iso]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/staging/fsl_qbman/fsl_usdpaa.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/staging/fsl_qbman/fsl_usdpaa.c b/drivers/staging/fsl_qbman/fsl_usdpaa.c
index ed986a0..9ef375e 100644
--- a/drivers/staging/fsl_qbman/fsl_usdpaa.c
+++ b/drivers/staging/fsl_qbman/fsl_usdpaa.c
@@ -877,7 +877,7 @@ out:
 		unsigned long longret;
 		unsigned long populate;
 		down_write(&current->mm->mmap_sem);
-		longret = do_mmap_pgoff(fp, 0, map->frag->len, PROT_READ |
+		longret = do_mmap_pgoff(fp, PAGE_SIZE, map->frag->len, PROT_READ |
 			(i->flags & USDPAA_DMA_FLAG_RDONLY ? 0 : PROT_WRITE),
 			MAP_SHARED, map->frag->pfn_base, &populate);
 		up_write(&current->mm->mmap_sem);
@@ -1023,11 +1023,9 @@ static int portal_mmap(struct file *fp, struct resource *res, void **ptr)
 	unsigned long populate;
 
 	down_write(&current->mm->mmap_sem);
-	do {
-		longret = do_mmap_pgoff(fp, 0, resource_size(res),
-					PROT_READ | PROT_WRITE, MAP_SHARED,
-					res->start >> PAGE_SHIFT, &populate);
-	} while (longret == 0);
+	longret = do_mmap_pgoff(fp, PAGE_SIZE, resource_size(res),
+				PROT_READ | PROT_WRITE, MAP_SHARED,
+				res->start >> PAGE_SHIFT, &populate);
 	up_write(&current->mm->mmap_sem);
 
 	if (populate)
-- 
1.8.4.93.g57e4c17

